unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/AvatarEditorHUDController.cs

```diff
@@ -359,6 +359,9 @@ public void LoadUserProfile(UserProfile userProfile, bool forceLoading)
         if (userProfile.avatar == null || string.IsNullOrEmpty(userProfile.avatar.bodyShape))
             return;
 
+        if (DataStore.i.exploreV2.nameOrDescriptionJustSaved.Get())
```

Comment:
im not really convinced on the solution since we are patching a kernel issue.. is it necessary after the kernel changes?

Reply:
That solves an issue that kernel is not handling. If you do the following steps, you will get some incosistencies>

1) Open avatar editor hud and change wearabales
2) Open the name tab while avatar editor hud is open with unsaved changes
3) Edit and save name/description
4) Since the save name triggers a profile update (which I think makes sense, to confirm that saving has been done correctly) the unsaved changes of the wearables get overridden.

This is the reason why I added that check. Another option would be to ask kernel just to send back the data that has been saved, but that may get messy on both sides.

Reply:
so probably instead of having a variable referring to saving the name or description of the profile, we could have an internal variable in `AvatarEditorHUDController` to check if the user has unsaved changes on the wearables. Then avoid replacing them if the profile gets updated in the middle? What do you think?

Reply:
Agreed; actually there was already a bool that had the dirty state of the avatar. So I just reused it.